domain/product/repository/postgres/postgres_category.go

```diff
@@ -0,0 +1,42 @@
+package postgres
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Mitra-Apps/be-store-service/domain/product/entity"
+	"github.com/Mitra-Apps/be-store-service/types"
+	"gorm.io/gorm"
+)
+
+func (p *Postgres) GetCategoriesByStoreIdRepo(ctx context.Context, input types.GetCategoriesByStoreIdRepoInput) (output types.GetCategoriesByStoreIdRepoOutput, err error) {
+	categories := []*entity.Category{}
+
+	productTable := "products"
+	productCategoryRelationsTable := "product_category_relations"
+	aliasCategoryTable := "\"categories\""
+	aliasProductTable := "\"products\""
+	aliasProductCategoryRelationsTable := "\"product_category_relations\""
+	selectCategoryColumn := fmt.Sprintf("%s.id, %s.name, %s.is_active", aliasCategoryTable, aliasCategoryTable, aliasCategoryTable)
+
+	tx := p.db.WithContext(ctx).
+		Select(selectCategoryColumn).
+		Joins(fmt.Sprintf(" INNER JOIN %s on %s.category_id = %s.id ", productCategoryRelationsTable, aliasProductCategoryRelationsTable, aliasCategoryTable)).
+		Joins(fmt.Sprintf(" INNER JOIN %s on %s.product_id = %s.id ", productTable, aliasProductCategoryRelationsTable, aliasProductTable)).
+		Where(fmt.Sprintf(" %s.store_id = ? ", aliasProductTable), input.StoreID).
+		Where(fmt.Sprintf("%s.is_active = ?", aliasCategoryTable), !input.IsIncludeDeactivated).
```

Comment:
this IsIncludeDeactivated, does it mean that if we include deactivated we should get all data for all data including both is_active true and false, or should only show is_active false only?

Reply:
yes, updated mas, if it is include deactivated, should be include all the data